inference-engine/src/inference_engine/src/cpp/ie_infer_request.cpp

```diff
@@ -202,20 +202,143 @@ InferRequest::InferRequest(const std::shared_ptr<void>& so, const ie::IInferRequ
     IE_ASSERT(_impl != nullptr);
 }
 
-void InferRequest::set_blob(const std::string& name, const ie::Blob::Ptr& data) {
-    INFER_REQ_CALL_STATEMENT(_impl->SetBlob(name, data);)
+void InferRequest::set_tensor(const std::string& name, const Tensor& tensor) {
+    INFER_REQ_CALL_STATEMENT(_impl->SetBlob(name, tensor._impl);)
 }
 
-ie::Blob::Ptr InferRequest::get_blob(const std::string& name) {
-    ie::Blob::Ptr blobPtr;
-    INFER_REQ_CALL_STATEMENT(blobPtr = _impl->GetBlob(name);)
-    std::string error = "Internal error: blob with name `" + name + "` is not allocated!";
-    const bool remoteBlobPassed = blobPtr->is<ie::RemoteBlob>();
-    if (blobPtr == nullptr)
-        IE_THROW() << error;
-    if (!remoteBlobPassed && blobPtr->buffer() == nullptr)
-        IE_THROW() << error;
-    return blobPtr;
+void InferRequest::set_input_tensor(const std::size_t idx, const Tensor& tensor) {
+    INFER_REQ_CALL_STATEMENT({
+        auto exec_network = _impl->GetExecutableNetwork();
+        IE_ASSERT(exec_network != nullptr);
+        auto func = exec_network->GetExecGraphInfo();
+        IE_ASSERT(func != nullptr);
+        _impl->SetBlob(func->get_parameters().at(idx)->get_friendly_name(), tensor._impl);
+    })
+}
+
+void InferRequest::set_input_tensor(const Tensor& tensor) {
+    set_input_tensor(0, tensor);
+}
+
+void InferRequest::set_output_tensor(const std::size_t idx, const Tensor& tensor) {
+    INFER_REQ_CALL_STATEMENT({
+        auto exec_network = _impl->GetExecutableNetwork();
+        IE_ASSERT(exec_network != nullptr);
+        auto func = exec_network->GetExecGraphInfo();
+        IE_ASSERT(func != nullptr);
+        auto source_output = func->get_results().at(idx)->get_input_source_output(0);
+        auto source_node = source_output.get_node();
+        auto name = source_node->get_friendly_name();
+        if (source_node->outputs().size() > 1) {
+            name += ("." + std::to_string(source_output.get_index()));
+        }
+        _impl->SetBlob(name, tensor._impl);
+    })
+}
```

Comment:
can we remove all such specializations like `set_output_tensor` etc and keep it for a separate PR?

Reply:
removed